packages/react/src/errorboundary.tsx

```diff
@@ -52,6 +64,43 @@ const INITIAL_STATE = {
   eventId: null,
 };
 
+/**
+ * Logs react error boundary errors to Sentry. If on React version >= 17, creates stack trace
+ * from componentStack param, otherwise relies on error param for stacktrace.
+ *
+ * @param error An error captured by React Error Boundary
+ * @param componentStack The component stacktrace
+ */
+function captureReactErrorBoundaryError(error: Error, componentStack: string): string {
+  const errorBoundaryError = new Error(error.name);
+  errorBoundaryError.stack = componentStack;
+  errorBoundaryError.message = error.message;
```

Comment:
Error constructor accepts a message, so you were effectively overriding it on line 77.

```suggestion
  const errorBoundaryError = new Error(error.message);
  errorBoundaryError.stack = componentStack;
```

Reply:
if you want to preserve `name `as well, you can do something along the lines of:

```js
const errorBoundaryError = new error.constructor(error.message);
errorBoundaryError.stack = componentStack;
```